File: iris-crypto/iris-crypto-rsa/src/main/java/city/smartb/iris/crypto/rsa/verifier/RS256Verifier.java
package city.smartb.iris.crypto.rsa.verifier;

import com.nimbusds.jose.JWSAlgorithm;

import java.security.GeneralSecurityException;
import java.security.Signature;
import java.security.interfaces.RSAPublicKey;

public class RS256Verifier implements Verifier {

	private final RSAPublicKey rsaPublicKey;

	public RS256Verifier(RSAPublicKey rsaPublicKey) {
		this.rsaPublicKey = rsaPublicKey;
	}

	@Override
	public JWSAlgorithm getAlgorithm() {
		return  JWSAlgorithm.RS256;
	}

	@Override
	public boolean verify(byte[] content, byte[] signature) throws GeneralSecurityException {
		Signature jcaSignature = Signature.getInstance("SHA256withRSA");
		jcaSignature.initVerify(rsaPublicKey);
		jcaSignature.update(content);
		return jcaSignature.verify(signature);
	}
}

File: iris-crypto/iris-crypto-key-rsa/src/main/kotlin/city/smartb/iris/crypto/rsa/verifier/RS256Verifier.kt
package city.smartb.iris.crypto.rsa.verifier

import city.smartb.iris.crypto.dsl.verifier.Verifier
import com.nimbusds.jose.JWSAlgorithm
import java.security.GeneralSecurityException
import java.security.Signature
import java.security.interfaces.RSAPublicKey

class RS256Verifier(
    private val rsaPublicKey: RSAPublicKey
) : Verifier {

    override val algorithm: JWSAlgorithm
        get() = JWSAlgorithm.RS256

    @Throws(GeneralSecurityException::class)
    override fun verify(content: ByteArray, signature: ByteArray): Boolean {
        val jcaSignature = Signature.getInstance("SHA256withRSA")
        jcaSignature.initVerify(rsaPublicKey)
        jcaSignature.update(content)
        return jcaSignature.verify(signature)
    }
}
